python/taichi/ui/gui.py

```diff
@@ -13,6 +14,21 @@ class Gui:
     def __init__(self, gui) -> None:
         self.gui = gui  #reference to a PyGui
 
+    @contextmanager
+    def sub_window(self, name, x, y, width, height):
+        """Creating a context manager for subwindow, all args are the same with `begin`.
```

Comment:
`all args are the same with begin` : since this is user facing we should copy paste the args over from `begin` and add an additional comment to developers saying "all args of this method should align with begin`. 
Alternatively how about we only expose this context manager to users? This is cleaner and easier for users? cc: @AmesingFlank

Reply:
Sounds good, let's remove `begin` and `end` and only keep `sub_window`. Although @Xuanwo , no need to worry about doing that in this PR, and many thanks for contributing!

Reply:
Additionally, removing `begin` and `end` and only keeping `sub_window` can help users prevent mismatched `begin`s and `end`s. It's a really good idea.

Reply:
I created an issue to track this idea, and I will fix it later.